### PR TITLE
Setup UI

### PR DESCRIPTION
**Pull request by Telematica**

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    ui:
+        container_name: 'ui'
+        build:
+          context: './ui'
+          args:
+            NODE_ENV: 'production'
+            REACT_APP_HOST_API: 'http://localhost:8080/api/v1'
+            REACT_APP_HOST_AUTH: 'http://localhost:10090/v1'
+        ports:
+          - '80:80'
```
Comment (pianodaemon) [inline docker-compose.yaml]: The output port for this stage should have been 8080

File: ui/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ENV GENERATE_SOURCEMAP=false
+RUN npm install react-scripts -g --silent
+RUN yarn install
+RUN yarn run build
+
+FROM mhart/alpine-node
+RUN yarn global add serve
+WORKDIR /app
+COPY --from=builder /app/build .
+CMD ["serve", "-p", "80", "-s", "."]doc
```
Comment (pianodaemon) [inline ui/Dockerfile:19]: What about the postfix doc ?

File: docker-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+version: '3.8'
+
+x-shared-environment: &commonEnv
+    REDIS_HOST: "cache"
+    REDIS_PORT: 6379
+
+services:
+    ui:
+        container_name: 'ui'
```
Comment (pianodaemon) [inline docker-compose.yaml]: "eastern_sea" <---- ?

**Review by pianodaemon — COMMENTED** [review]
I left a few comments

File: ui/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './index.css';
+import App from './App';
+import reportWebVitals from './reportWebVitals';
+
+ReactDOM.render(
+  <React.StrictMode>
+    <App />
```
Comment (pianodaemon) [inline ui/src/index.tsx:9]: Interesting it is liike a xhtml tag macro devised by you

File: ui/src/App.test.tsx [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import App from './App';
+
+test('renders learn react link', () => {
```
Comment (pianodaemon) [inline ui/src/App.test.tsx:5]: Intriguing  :)
I've never imagined how render of an element could be tested

File: docker-compose.yaml (hunk 3 above)
Comment (pianodaemon) [reply docker-compose.yaml]: This file is not longer required as part of your commit

File: docs/Alphaville.txt [hunk 6]
```diff
@@ -0,0 +1,6 @@
+
+Dilca
```
Comment (pianodaemon) [inline docs/Alphaville.txt:2]: is this file strictly needed ?

File: ui/Dockerfile (hunk 2 above)
Comment (Telematica) [reply ui/Dockerfile:19]: My mistake

File: ui/src/index.tsx (hunk 4 above)
Comment (Telematica) [reply ui/src/index.tsx:9]: Hahaha, it enforces JS strict mode, which is cool (almost) to have it enable...

File: ui/src/App.test.tsx (hunk 5 above)
Comment (Telematica) [reply ui/src/App.test.tsx:5]: All is simulated via a JS engine that acts nearly as actual Web browser render engine.

File: docs/Alphaville.txt (hunk 6 above)
Comment (Telematica) [reply docs/Alphaville.txt:2]: My notes on last conversation with Gerardo, we can drop it.
